src/cpp/xmlparser/XMLParser.cpp

```diff
@@ -1462,6 +1478,160 @@ XMLP_ret XMLParser::parseDynamicTypes(tinyxml2::XMLElement* p_root)
+        }
+        else if (std::strcmp(classStr.c_str(), "FileConsumer") == 0)
+        {
+            std::string outputFile = "output.log";
+            bool append = false;
+
+            tinyxml2::XMLElement* property = consumer.FirstChildElement(PROPERTY);
+            if (nullptr == property)
+            {
+                std::unique_ptr<FileConsumer> consumer(new FileConsumer);
+                Log::RegisterConsumer(std::move(consumer));
+            }
+            else
+            {
+                tinyxml2::XMLElement* p_auxName = nullptr;
+                tinyxml2::XMLElement* p_auxValue = nullptr;
+                while (nullptr != property)
+                {
+                    // name - stringType
+                    if (nullptr != (p_auxName = property->FirstChildElement(NAME)))
+                    {
+                        std::string s = p_auxName->GetText();
+
+                        if (std::strcmp(s.c_str(), "filename") == 0)
+                        {
+                            if (nullptr != (p_auxValue = property->FirstChildElement(VALUE)))
+                            {
+                                outputFile = p_auxValue->GetText();
+                            }
+                            else
+                            {
+                                logError(XMLParser, "Filename value cannot be found for " << classStr
+                                    << " log consumer.");
+                            }
+                        }
+                        else if (std::strcmp(s.c_str(), "append") == 0)
+                        {
+                            if (nullptr != (p_auxValue = property->FirstChildElement(VALUE)))
+                            {
+                                std::string auxBool = p_auxValue->GetText();
+                                if (std::strcmp(auxBool.c_str(), "TRUE") == 0)
+                                {
+                                    append = true;
+                                }
+                            }
+                            else
+                            {
+                                logError(XMLParser, "Append value cannot be found for " << classStr
+                                    << " log consumer.");
+                            }
+                        }
+                        else
+                        {
+                            logError(XMLParser, "Unknown property " << s << " in " << classStr
+                                << " log consumer.");
+                        }
+                    }
+                    property = property->NextSiblingElement(PROPERTY);
+                }
+
+                std::unique_ptr<FileConsumer> consumer(new FileConsumer(outputFile, append));
+                Log::RegisterConsumer(std::move(consumer));
+            }
+        }
+        else
+        {
+            logError(XMLParser, "Unknown log consumer class: " << classStr);
```

Comment:
Also return an error when an error occur in all this code. Use a variable to store the returned value instead of a lot of `return` sentences.

Reply:
Updated following your suggestions